libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -55,7 +55,7 @@
 extern const AP_HAL::HAL &hal;
 
 // baudrates to try to detect GPSes with
-const uint32_t AP_GPS::_baudrates[] = {4800U, 19200U, 38400U, 115200U, 57600U, 9600U, 230400U};
+const uint32_t AP_GPS::_baudrates[] = {9600U, 115200U, 9600U, 115200U, 4800U, 19200U, 38400U, 57600U, 230400U};
```

Comment:
Why is 9600U here twice?

Reply:
@peterbarker wrote in the PR description that it now probes twice 9600 and 115200. That doesn't seem good to me; why is that needed?

Reply:
In case the GPS isn't ready the first time we ask it at the most probable baudrates.

Reply:
Do you have any log that shows this ever actually occuring? The GPS should have always beaten us on boot time and being ready to listening to the serial port. Doubling up here just slows down other baud rates/changes so I'd prefer to not do it on spec.